### ISSUE TITLE
[Error] Call to a member function toArray() on array  

### ISSUE THREAD

**Issue by hadouke** [facebook/php-webdriver#642]
### What are you trying to achieve? (Expected behavior)
Working tests with Codeception 2.5.6 + WebDriver module in pair with npm webdriver-manager + facebook/webdriver inside a Chrome Browser

### What do you get instead? (Actual behavior)
After update composer dependencies with `facebook/webdriver 1.6 > 1.7` - I encounter an error on test\suite start even before browser is up.
                                                        
```
  [Error] Call to a member function toArray() on array  
                                                        
#1  /var/lib/jenkins/workspace/project/testing/vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php:112
```

### How could the issue be reproduced? (Steps to reproduce)
Configure: 
Acceptance yaml configuration to run Chrome is 👍 
```
    - WebDriver:
        url: 'https://localhost/'
        wait: 1
        browser: chrome
        window_size: false
        capabilities:
          chromeOptions:
            args: ["--disable-gpu", "--window-size=1920,2500"] #"--headless",

```
Test example is Cest format:
```
public function testMe(AcceptanceTester $I) {
$I->amOnPage('/login');
}
```

### Details

```
composer.json:
require: {
        "facebook/webdriver": "1.6",
        "codeception/codeception": "^2.4",
}
```
* Php-webdriver version: 1.7
* PHP version: 7.1
* Selenium server version: selenium-server-standalone-3.141.59.jar
* Operating system: Windows 10 (1809)
* Browser used + version: Chrome Version 75.0.3770.80 (Official Build) (64-bit)  chromedriver_75.0.3770.8

**Comment by Lctrs** [comment]
This happens if the same instance of a `DesiredCapabilities` object is passed twice or more to the `RemoteWebDriver::create()` function. The first call will transform the `ChromeOptions` into an array on the same key in the internal array of the `DesiredCapabilities` object. So, on the second call, `$desired_capabilities->getCapability(ChromeOptions::CAPABILITY);` will return the transformed array of the first call.

Adding `$desired_capabilities = clone $desired_capabilities;` before initializating the `WebDriverCommand` on line 136 in `RemoteWebDriver` fixed the issue for me.

**Comment by mstaatz** [comment]
In my opinion the referenced PR is not the solution... the Error is in Line 112 as you can see in the the given trace.

Problematic Code on line 112:
`if (!isset($chromeOptions->toArray()['w3c'])) `
`$chromeOptions` can be an Array or an Object because of the code in line 109 and 110:
`$currentChromeOptions = $desired_capabilities->getCapability(ChromeOptions::CAPABILITY);
            $chromeOptions = !empty($currentChromeOptions) ? $currentChromeOptions : new ChromeOptions();` 

i have the same issue and in my case `$currentChromeOptions` is an array and this is why the error occure

**Comment by Lctrs** [comment]
@mstaatz The error come from `DesiredCapabilities#toArray()` that mutates its internal array and transform the value at the key `ChromeOptions::CAPABILITY` into an array. So subsequent call to `DesiredCapabilities#getCapability(ChromeOptions::CAPABILITY)` will return an array. Hence the error.

The referenced PR does fix the error. But we also can fix `DesiredCapabilities#toArray()` to turn it to an immutable operation.

**Comment by OndraM** [comment]
https://github.com/facebook/php-webdriver/pull/643#issuecomment-500789115 - and see the updated PR. This should IMO fix this and not introduce any change in the behavior.

**Comment by mstaatz** [comment]
@Lctrs ... did not fix for me ... that's why i writing here. For me all works fine without:

`if (!isset($chromeOptions->toArray()['w3c'])) {
    $chromeOptions->setExperimentalOption('w3c', false);
}`

**Comment by Lctrs** [comment]
@mstaatz Maybe you have a piece of code elsewhere that call `DesiredCapabilities#toArray()`. I don't know. Anyway, @OndraM has provided a more robust fix for this.

**Comment by hadouke** [comment]
> @Lctrs ... did not fix for me ... that's why i writing here. For me all works fine without:
> 
> `if (!isset($chromeOptions->toArray()['w3c'])) { $chromeOptions->setExperimentalOption('w3c', false); }`

Provided hot fix, with 
`$desired_capabilities = clone $desired_capabilities;`
 at line 134, after executor is set, not helped as well. 

Only commenting 
`            if (!isset($chromeOptions->toArray()['w3c'])) {
                $chromeOptions->setExperimentalOption('w3c', false);
            }
`
will start successfully new browser session. But here we come with another problem, that any type of `click` stop working.

**Comment by OndraM** [comment]
@hadouke Please see the updated PR in #643 .

**Comment by OndraM** [comment]
Merge to community branch. You can try it using "dev-community" dependency in composer.json.

Also 1.7.1 will be tagged soon.

**Comment by matvey88** [comment]
@OndraM thanks! Works for me, in "dev-community" dependency🙏🏾

**Comment by javjim** [comment]
Works for me too in dev-community

**Comment by OndraM** [comment]
Released in 1.7.1.
